salt/runner.py

```diff
@@ -301,9 +301,10 @@ def _run_runner(self):
             if not self.opts.get('quiet', False):
                 display_output(ret, 'nested', self.opts)
         else:
-            ret = {
-                'retcode': salt.defaults.exitcodes.EX_SOFTWARE,
-            }
+            if not ret:
+                ret = {
+                    'retcode': salt.defaults.exitcodes.EX_SOFTWARE,
+                }
```

Comment:
I'd have to look into this to understand the correct behavior - but if I understand things correctly this would skip the `EX_SOFTWARE` retcode if `ret` already exists. Is that intentional? Would it be possible to add some comment to this code explaining why that should be the case?

Reply:
if ret isn't an empty dict. I think this is a bug introduced at some point in the recent past, as with try, except, else;  the else is always fired. as such it was discarding the ret. i suspect noone noticed because runnerclient is strictly for cli use (runner is for netapi), as such run_plus_plus in the test suite is the only thing actually invoking it and relying on the return in this way.

Reply:
fwiw this change is unrelated to PR, I just made it to fix the failing tests